### ISSUE TITLE
Bazel support

### ISSUE THREAD

**Issue by xinxiao** [google/guice#1227]
Could there be better support for bazel?

**Comment by Vincent-M** [comment]
Are you asking about including the Guice's `.jar` in your Bazel project?

To get Guice to work in Bazel, you can simply use the jar like some [examples](https://gerrit.googlesource.com/gerrit/+/master/WORKSPACE#106) show:

```
maven_jar(
    name = "guice_library",
    artifact = "com.google.inject:guice:4.2.2,
    sha1 = "6dacbe18e5eaa7f6c9c36db33b42e7985e94ce77",
)
```
You targets can then specify  "@guice_library//jar" as a dependency to depend on this jar.

However it seems that maven_jar is not future-proof, as it is [apparently being deprecated](https://github.com/bazelbuild/bazel/issues/1752#issuecomment-475324296):

[Bazel documentation](https://docs.bazel.build/versions/master/external.html#repository-rules) recommends using `http_archive` instead, not sure how this would be replaced

**Comment by xinxiao** [comment]
@Vincent-M Hi, thanks for addressing this. I found current bazel Guice import not simple is because other than importing the Guice core, bazel requires me to import additional dependencies manually. I tried to import Guice through [Guice Parent](https://mvnrepository.com/artifact/com.google.inject/guice-parent/4.2.2), but this is a pom path rather than a jar import, which is hard to handle in bazel. A comprehensive maven jar route for Guice could be handy in my use case.
